### PR TITLE
`azurerm_new_relic_monitor` - Support `identity`

### PR DESCRIPTION
**Pull request by ms-zhenhua**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description

Support `identity` for `azurerm_new_relic_monitor` to enable `metrics flow` like [data dog](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/datadog_monitors#role-assignment)

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

image

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_new_relic_monitor` - Support `identity`

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/newrelic/new_relic_monitor_resource.go [hunk 1]
```diff
@@ -520,3 +532,34 @@ func flattenUserInfoModel(input *monitors.UserInfo) []UserInfoModel {
 
 	return append(outputList, output)
 }
+
+// Currently the API only supports `SystemAssigned` type.
```
Comment (stephybun) [inline internal/services/newrelic/new_relic_monitor_resource.go:536]: If the Swagger spec specifies that System and User assigned should be supported but the API in reality only supported SystemAssigned then we should raise an issue on the Rest API specs repo.

File: internal/services/newrelic/new_relic_monitor_resource.go [hunk 2]
```diff
@@ -265,6 +268,11 @@ func (r NewRelicMonitorResource) Create() sdk.ResourceFunc {
 				},
 			}
 
+			identityValue := expandSystemAssigned(metadata.ResourceData.Get("identity").([]interface{}))
+			if identityValue.Type != identity.TypeNone {
+				properties.Identity = identityValue
+			}
```
Comment (stephybun) [inline internal/services/newrelic/new_relic_monitor_resource.go:274]: This isn't how the `Type` field in `Identity` is usually handled which would imply a bug, this should also be raised as an issue on the Rest API specs repo.

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @ms-zhenhua. It looks like there are some fundamental issues with the `identity` property for this resource that should be raised with the Service Team, could you please open a swagger issue regarding the comments I left in-line and then link those in the code?

File: internal/services/newrelic/new_relic_monitor_resource.go (hunk 2 above)
Comment (ms-zhenhua) [reply internal/services/newrelic/new_relic_monitor_resource.go:276]: created a issue to track this problem:  https://github.com/Azure/azure-rest-api-specs/issues/29257

File: internal/services/newrelic/new_relic_monitor_resource.go (hunk 1 above)
Comment (ms-zhenhua) [reply internal/services/newrelic/new_relic_monitor_resource.go]: created a issue to track this problem: https://github.com/Azure/azure-rest-api-specs/issues/29256

**Comment by ms-zhenhua** [comment]
Hi @stephybun, thank you for your review. I have created the issues and added them to the comments. Please take another review.

**Review by stephybun — APPROVED** [review]
Thanks @ms-zhenhua LGTM 👍

File: internal/services/newrelic/new_relic_monitor_resource.go (hunk 2 above)
Comment (tombuildsstuff) [reply internal/services/newrelic/new_relic_monitor_resource.go:276]: In the future: can we also update the API Definition to expose the `Identity` field correctly too? We've spent a _considerable_ amount of time standardising the behaviours (and methods) for the `Identity` types, so we should lean on them where possible - as such I've opened https://github.com/hashicorp/pandora/pull/4171 which'll allow this.

**Pull request merged**